Sources/Sentry/SentryTransaction.m

```diff
@@ -54,6 +56,11 @@ - (instancetype)initWithTrace:(SentryTracer *)trace children:(NSArray<id<SentryS
         serializedData[@"_metrics_summary"] = metricsSummary;
         [serializedTrace removeObjectForKey:@"_metrics_summary"];
     }
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+    NSMutableDictionary *traceProfileEntry = [NSMutableDictionary dictionary];
+    traceProfileEntry[@"profile_id"] = self.trace.profileSessionID;
+    serializedTrace[@"data"] = traceProfileEntry;
```

Comment:
`h`: This overwrites existing fields in the `data` keyword. Instead, we must add the `profile_id` to the existing data dictionary. I'm surprised that no test fails for this.

Reply:
good catch. i wrote a test that failed, and then fixed the issue.